norfair/lib_metrics.py

```diff
@@ -0,0 +1,376 @@
+    else:
+        videos = arguments[2:]
+
+    return videos, input_path, save_path, make_video, show_metrics, save_pred
+
+
+class text_file:
+    def __init__(self, input_path=None, save_path=".", file_name=None):
+
+        if input_path is None:
+            raise ValueError(
+                "You must set 'input_path' argument when setting 'text_file' class"
+            )
+        if file_name is None:
+            raise ValueError(
+                "You must set 'file_name' argument when setting 'text_file' class"
+            )
+
+        predictions_folder = os.path.join(save_path, "predictions")
+        if not os.path.exists(predictions_folder):
+            os.makedirs(predictions_folder)
+
+        out_file_name = os.path.join(predictions_folder, file_name + ".txt")
+        self.text_file = open(out_file_name, "w+")
+
+        self.frame_number = 1
+
+    def update_text_file(self, predictions):
+        write_predictions(
+            frame_number=self.frame_number, objects=predictions, out_file=self.text_file
+        )
+        self.frame_number += 1
+
+    def close_file(self):
+        self.text_file.close()
+
+
+class Det_from_file:
+    def __init__(self, input_path=None, file_name=None):
+        # get detecions matrix data with rows corresponding to:
+        # frame, id, bb_left, bb_top, bb_right, bb_down, conf, x, y, z
+        if input_path is None:
+            raise ValueError(
+                "You must set 'input_path' argument when setting 'Det_from_file' class"
+            )
+        if file_name is None:
+            raise ValueError(
+                "You must set 'file_name' argument when setting 'Det_from_file' class"
+            )
+        detections_path = os.path.join(input_path, file_name, "det/det.txt")
+        self.matrix_detections = Process_text_file(path=detections_path)
+
+    def get_dets_from_frame(self, frame_number=None):
+        # this function returns a list of norfair Detections class, corresponding to frame=frame_number
+
+        if frame_number is None:
+            raise ValueError(
+                "You must set 'frame_number' argument when calling get_dets_from_frame()"
+            )
+
+        indexes = np.argwhere(self.matrix_detections[:, 0] == frame_number)
+        detections = []
+        if len(indexes) > 0:
+            actual_det = self.matrix_detections[indexes]
+            actual_det.shape = [actual_det.shape[0], actual_det.shape[2]]
+            for j in range(len(indexes)):
+                points = [actual_det[j, [2, 3]], actual_det[j, [4, 5]]]
+                points = np.array(points)
+                conf = actual_det[j, 6]
+                new_detection = Detection(
+                    points, np.array([1, 1])
+                )  # set to [1,1] or [conf,conf]
+                detections.append(new_detection)
+        self.actual_detections = detections
+        return detections
+
+
+class accumulators:
+    def __init__(self, input_path=None):
```

Comment:
Describe what `input_path` refers to, it's not clear from first reading the class.